pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -396,6 +528,14 @@ func (pl *DynamicResources) PreFilter(ctx context.Context, state fwk.CycleState,
 		} else {
 			allocateClaims = append(allocateClaims, claim)
 
+			if pl.enableExtendedResource {
+				// continue without validating the special claim for extended resource backed by DRA
+				// when it is not allocated yet at this point, and it does not have a spec.
+				if claim.UID == s.extendedResourceClaimTplUID {
+					continue
+				}
+			}
```

Comment:
Does the code become simpler when not extending `claims` here? Then you don't need to skip the special entry and the need for `extendedResourceClaimTplUID` might go away. 

You can then extend `claims` and `claimInfos` in `Reserve`. Then all code which follows works as before, by iterating over `claims`.

Reply:
at Filter phase line: 779, 
			unavailableClaims = append(unavailableClaims, index)

unavailableClaims may need to store the special claim backed by DRA (it was persisted from the prior scheduling cycle), then this scheduling cycle checks if the node still fits for the claim.

this would require the special claim backed by DRA has an index in the state.claims, i.e., state.claims need to have the special claim backed by DRA

Reply:
We can use -1 as special index to indicate "special ResourceClaim". The code which consumes these indices has to have a special case for it anyway.